quarkus/config-api/src/main/java/org/keycloak/config/WildcardOptionsUtil.java

```diff
@@ -0,0 +1,121 @@
+package org.keycloak.config;
+
+/**
+ * Utility class for working with configuration options that use wildcards.
+ * <p>
+ * Wildcard options are configuration keys that contain a variable segment enclosed between {@link #WILDCARD_START} and {@link #WILDCARD_END} characters.
+ * <p>
+ * Wildcard options in Keycloak always <strong>end</strong> with the variable segment.
+ */
+public class WildcardOptionsUtil {
+
+    /**
+     * Marker indicating the start of a wildcard segment in a configuration key.
+     */
+    public static final String WILDCARD_START = "<";
+
+    /**
+     * Marker indicating the end of a wildcard segment in a configuration key.
+     */
+    public static final String WILDCARD_END = ">";
+
+        /**
+     * Determines whether the given configuration key represents a wildcard option.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * isWildcardOption("tracing-header-<header>")      → "true"
+     * isWildcardOption("tracing-header-<headxxx")      → "false"
+     * isWildcardOption("tracing-header-headxxx>")      → "false"
+     * isWildcardOption("db-kind-<datasource>")         → "true"
+     * isWildcardOption("http-port")                    → "false"
+     * isWildcardOption("quarkus.<sth>.end")            → "false"
+     * }</pre>
+     *
+     * @param key the configuration key to check
+     * @return {@code true} if the key represents a Keycloak wildcard option
+     */
+    public static boolean isWildcardOption(String key) {
+        return key != null && key.contains(WILDCARD_START) && key.endsWith(WILDCARD_END);
+    }
+
+    /**
+     * Determines whether the given configuration key represents an external wildcard option (variable segment does not have to be at the end)
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * isExternalWildcardOption("tracing-header-<header>")      → "true"
+     * isExternalWildcardOption("tracing-header-<headxxx")      → "false"
+     * isExternalWildcardOption("tracing-header-headxxx>")      → "false"
+     * isExternalWildcardOption("db-kind-<datasource>")         → "true"
+     * isExternalWildcardOption("http-port")                    → "false"
+     * isExternalWildcardOption("quarkus.<sth>.end")            → "true"
+     * }</pre>
+     *
+     * @param key the configuration key to check
+     * @return {@code true} if the key represents an external wildcard option
+     */
+    public static boolean isExternalWildcardOption(String key) {
```

Comment:
I'd be maybe more explicit in the naming here. Maybe `isKcOrQuarkusWildcardOption`? "External" would suggest we're looking for Quarkus options only.